offchain/types/src/blockchain_config.rs

```diff
@@ -97,31 +97,52 @@ impl TryFrom<BlockchainCLIConfig> for BlockchainConfig {
 
     fn try_from(cli: BlockchainCLIConfig) -> Result<Self, Self::Error> {
         // read the files
-        let dapp: Contract = read(cli.dapp_deployment_file)?;
-        let rollups: RollupsDeployment = read(cli.rollups_deployment_file)?;
+        let dapp =
+            cli.dapp_deployment_file.map(read::<Contract>).transpose()?;
+        let rollups = cli
+            .rollups_deployment_file
+            .map(read::<RollupsDeployment>)
+            .transpose()?;
 
         // try to get the values from the environment values
         // default to the files
-        let dapp_address =
-            env_or_file(cli.dapp_address, dapp.address, "dapp_address")?;
+        let dapp_address = env_or_file(
+            cli.dapp_address,
+            dapp.clone().map(|file| file.address).flatten(),
```

Comment:
The only way I could find to remove this clone is by using `Option.take`, but it is unsafe.
https://doc.rust-lang.org/std/option/enum.Option.html#method.take

Reply:
I did it, but it just gets worse.